include/kernels/spinor_noise.cuh

```diff
@@ -0,0 +1,60 @@
+#include <math_helper.cuh>
+#include <color_spinor_field_order.h>
+#include <random_helper.h>
+#include <kernel.h>
+
+namespace quda {
+
+  using namespace colorspinor;
+
+  template <typename real_, int nSpin_, int nColor_, QudaFieldOrder order, QudaNoiseType noise_>
+  struct SpinorNoiseArg : kernel_param<> {
+    using real = real_;
+    static constexpr int nSpin = nSpin_;
+    static constexpr int nColor = nColor_;
+    static constexpr QudaNoiseType noise = noise_;
+    using V = typename colorspinor::FieldOrderCB<real, nSpin, nColor, 1, order>;
+    V v;
+    RNGState *rng;
+    SpinorNoiseArg(ColorSpinorField &v, RNGState *rng) :
+      kernel_param(dim3(v.VolumeCB(), v.SiteSubset(), 1)),
+      v(v),
+      rng(rng) { }
+  };
+
+  template<typename real, typename Arg> // Gauss
+  __device__ __host__ inline void genGauss(Arg &arg, RNGState& localState, int parity, int x_cb, int s, int c) {
+    real phi = 2.0*M_PI*uniform<real>::rand(localState);
+    real radius = uniform<real>::rand(localState);
+    radius = sqrt(-log(radius));
```

Comment:
Do you think it's worth quickly implementing a custom dispatch for `sqrt` (which uses fast rsqrt and multiply) and `log` in `math_helper.cuh`?

Reply:
I don't think I'd want to use that here since it would be too dangerous for argument equal to zero.

Reply:
Don't we have bigger problems, then? Or is it because we'd get a silent issue from the fast version?